src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1017,6 +1018,8 @@ private ImmutableArray<TypeWithAnnotations> SubstituteTypeArguments(ImmutableArr
             {
                 Debug.Assert(realTypeArguments.Length == 0);
             }
+
+            receiver ??= new BoundTypeExpression(syntax, null, synthesizedMethod.ContainingType);
```

Comment:
which cases are handled by doing this here instead of initializing it after checking `closureKind == ClosureKind.Static`?